pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafOperator.java

```diff
@@ -714,15 +716,32 @@ public long merge(long value1, long value2) {
       public String getStatName() {
         return "responseSerializationCpuTimeNs";
       }
-    };
+    },
+    /**
+     * Allocated memory in bytes for this operator or its children in the same stage.
+     */
+    ALLOCATED_MEMORY_BYTES(StatMap.Type.LONG, null),
+    /**
+     * Time spent on GC while this operator or its children in the same stage were running.
+     */
+    GC_TIME_MS(StatMap.Type.LONG, null);
+    // IMPORTANT: When adding new StatKeys, make sure to either create the same key in BrokerResponseNativeV2.KeyStat or
+    //  call the constructor that accepts a String as last argument and set it to null.
+    //  Otherwise the constructor will fail with an IllegalArgumentException which will not be caught and will
+    //  propagate to the caller, causing the query to timeout.
```

Comment:
Why will the query timeout?

Reply:
Because the exception won't be caught, it will end up killing the task that executed the opchain. This means that this opchain will never send an EOS block, which in turn means that the parent tasks will never finish. Consequently, the query will end up timing out.

I was in fact affected by this issue when testing this PR, and I didn't find a way to fail in a better way, so I decided to add this commen,t hoping this will help future contributors (or a future me :laughing: )

Reply:
Ah I see, the uncaught exception would be propagated from the enum constructor and since enums are singletons that are initialized during the enum classloading, I think the exception would be wrapped inside an `ExceptionInInitializerError` which won't be caught (and the enum type won't be loaded properly IIUC).